src/ratelimitedfunc.js

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+module.exports = function(f, min_interval_ms) {
+    this.lastCall = 0;
+    this.scheduledCall = undefined;
+
+    var self = this;
+    return function() {
+        var now = Date.now();
+
+        if (self.lastCall < now - min_interval_ms) {
+            f.apply(this);
+            self.lastCall = now;
+        } else if (self.scheduledCall === undefined) {
+            self.scheduledCall = setTimeout(() => {
+                    self.scheduledCall = undefined;
+                    f.apply(this);
+                    self.lastCall = now;
+                },
+                (self.lastCall + min_interval_ms) - now
+            );
```

Comment:
Not sure about the indentation here. I was expecting to see a `}` on this line from the `setTimeout` function definition. It's not immediately obvious which things are open/closed with fat arrow syntax like `(() => {` , hence I would prefer:

```javascript
self.scheduledCall = setTimeout(() => {
    // stuff
}, (self.lastCall + min_interval_ms) - now);
```
or (but this is much more whitespace :( ):
```javascript
self.scheduledCall = setTimeout(
    () => {
        // stuff
    },
    (self.lastCall + min_interval_ms) - now
);
```